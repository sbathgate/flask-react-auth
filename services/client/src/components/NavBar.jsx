import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./NavBar.css";

const titleStyle = {
  fontWeight: "bold"
};

const NavBar = props => (
  <nav
    className="navbar is-dark"
    role="navigation"
    aria-label="main navigation"
  >
    <section className="container">
      <div className="navbar-brand">
        <Link to="/" className="navbar-item nav-title" style={titleStyle}>
          {props.title}
        </Link>
        <span
          className="nav-toggle navbar-burger"
          onClick={() => {
            let toggle = document.querySelector(".nav-toggle");
            let menu = document.querySelector(".navbar-menu");
            toggle.classList.toggle("is-active");
            menu.classList.toggle("is-active");
          }}
        >
          <span />
          <span />
          <span />
        </span>
      </div>
      <div className="navbar-menu">
        <div className="navbar-start">
          <Link to="/about" className="navbar-item">
            About
          </Link>
          <Link to="/status" className="navbar-item">
            User Status
          </Link>
        </div>
        <div className="navbar-end">
          <Link to="/register" className="navbar-item">
            Register
          </Link>
          <Link to="/login" className="navbar-item">
            Log In
          </Link>
          <span
            // eslint-disable-next-line react/jsx-handler-names
            onClick={props.logoutUser}
            className="navbar-item link"
          >
            Log Out
          </span>
        </div>
      </div>
    </section>
  </nav>
);

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
  logoutUser: PropTypes.func.isRequired
};

export default NavBar;
