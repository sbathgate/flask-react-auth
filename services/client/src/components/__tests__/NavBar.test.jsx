import React from "react";
import { cleanup } from "@testing-library/react";

import NavBar from "../NavBar";

afterEach(cleanup);

const props = {
  title: "Hello, World!",
  logoutUser: () => {
    return true;
  }
};

it("renders a title", () => {
  const { getByText } = renderWithRouter(<NavBar {...props} />);
  expect(getByText(props.title)).toHaveClass("nav-title");
});

it("renders", () => {
  const { asFragment } = renderWithRouter(<NavBar {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
