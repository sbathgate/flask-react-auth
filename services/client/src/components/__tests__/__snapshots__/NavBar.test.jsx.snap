// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    aria-label="main navigation"
    class="navbar is-dark"
    role="navigation"
  >
    <section
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item nav-title"
          href="/"
          style="font-weight: bold;"
        >
          Hello, World!
        </a>
        <span
          class="nav-toggle navbar-burger"
        >
          <span />
          <span />
          <span />
        </span>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            href="/about"
          >
            About
          </a>
          <a
            class="navbar-item"
            href="/status"
          >
            User Status
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <a
            class="navbar-item"
            href="/register"
          >
            Register
          </a>
          <a
            class="navbar-item"
            href="/login"
          >
            Log In
          </a>
          <span
            class="navbar-item link"
          >
            Log Out
          </span>
        </div>
      </div>
    </section>
  </nav>
</DocumentFragment>
`;
